lib/login.rb

```diff
@@ -38,19 +29,34 @@ def new_token_by_password(user:, password:, env:)
       end
 
       unless res[:success]
-        msg = "Error getting bearer token, see log"
-        if res[:error]
-          raise res[:error] rescue raise msg rescue e=$!
-        else
-          raise msg rescue e=$!
+        begin
+          msg = "Error getting bearer token, see log"
+          if res[:error]
+            raise res[:error] rescue raise msg rescue e=$!
+          else
+            raise msg rescue e=$!
+          end
+          Http.logger.error(e) # default error printing exclude cause
+          raise e
+        ensure
+          debug_try_get_console(env: env)
         end
-        Http.logger.error(e) # default error printing exclude cause
-        raise e
       end
 
       return res[:token], res[:valid_until]
     end
 
+
+    # Try HTTP GET console route for debugging when Auth error
+    # if console_url is accessible then it is likely Auth issue
+    # if console_url is inaccessible as well then likely network related issue including ingress, snd or platform glitch
+    def debug_try_get_console(env:)
+      console_url = env.opts[:admin_console_url] || env.api_endpoint_url.delete_suffix(':6443').gsub(/api/, 'console-openshift-console.apps')
```

Comment:
can we use `env.admin_console_url` instead of `env.api_endpoint_url.delete_suffix(':6443').gsub(/api/, 'console-openshift-console.apps')`?